src/Widgets/calendar/GridDay.vala

```diff
@@ -82,6 +89,38 @@ public class DateTime.Widgets.GridDay : Gtk.EventBox {
     public void update_date (GLib.DateTime date) {
         this.date = date;
         label.label = date.get_day_of_month ().to_string ();
+
+        Widgets.CalendarModel.get_default ().events_added.connect (update_event_days);
+        Widgets.CalendarModel.get_default ().events_updated.connect (update_event_days);
+        Widgets.CalendarModel.get_default ().events_removed.connect (update_event_days);
+    }
+
+    public async void update_event_days () {
+        GLib.Idle.add (() => {
+            var events = Widgets.CalendarModel.get_default ().get_events (date);
+            var event_dot_grid = new Gtk.Grid ();
+            event_dot_grid.column_homogeneous = true;
+            if (event_dot_grid != null) {
+                event_dot_grid.destroy ();
+            }
+            if (events.size != 0) {
+                foreach (var e in events) {
+                    if (e != null) {
+                        var event_dot = new Gtk.Image.from_icon_name ("pager-checked-symbolic", Gtk.IconSize.MENU);
+                        event_dot.visible = true;
```

Comment:
Is this necessary?

Reply:
Yes, without it, the dot doesn't show. I've tried it without those:
* The icon used is pager-checked-symbolic, to actually draw the dot.
* It's set to `visible = true`, because for some reason it's set to false at Gtk drawing startup or something.